electron/main.js

```diff
@@ -111,27 +120,20 @@ const createTray = () => {
   const trayPath = getUpdatedTrayIcon(
     isWindows || isMac ? TRAY_ICONS.LOGGED_OUT : TRAY_ICONS_PATHS.LOGGED_OUT,
   );
-  const tray = new Tray(trayPath);
+  tray = new Tray(trayPath);
 
   const contextMenu = Menu.buildFromTemplate([
     {
       label: 'Show app',
-      click: function () {
-        mainWindow.show();
-      },
+      click: () => mainWindow?.show(),
     },
     {
       label: 'Hide app',
-      click: function () {
-        mainWindow.hide();
-      },
+      click: () => mainWindow?.hide(),
     },
     {
       label: 'Quit',
-      click: async function () {
-        await beforeQuit();
```

Comment:
removed this as `app.quit` emits before-quit event, which is caught in app.on('before-quit') and calls beforeQuit again